mollie-payments-for-woocommerce/includes/mollie/wc/gateway/abstract.php

```diff
@@ -1278,9 +1265,22 @@ public function getReturnRedirectUrlForOrder( WC_Order $order ) {
 
 			}
 
-			$payment = Mollie_WC_Plugin::getPaymentObject()->getActiveMolliePayment($order_id, false );
+            try {
+                $payment = $this->activePaymentObject($orderId, false);
+            } catch (UnexpectedValueException $exc) {
+                Mollie_WC_Plugin::addNotice(
+                    __(
+                        'There was a problem when processing your payment. Please try again.',
+                        'mollie-payments-for-woocommerce'
+                    )
+                );
+                // Return to order payment page
+                if (method_exists($order, 'get_checkout_payment_url')) {
```

Comment:
Why is this check necessary here? Is this a case of features being changed in later versions of the lib?

Reply:
@XedinUnknown Because the plugin support WooCommerce 2.2.x I copied the solution from another place see line 1253.